modules/sagemaker/sagemaker-model-cicd/lib/code-pipeline/model-deploy/model-deploy-code-pipeline-stack.ts

```diff
@@ -0,0 +1,227 @@
+  readonly infraRepo: Repository;
+  readonly deploymentGroup: DeploymentGroup;
+  readonly ssmParamName?: string;
+  readonly modelPackageGroupName: string;
+  readonly sagemakerArtifactsBucketName: string;
+}
+
+export class ModelDeployCodePipelineStack extends cdk.Stack {
+  public readonly pipeline: cdk.pipelines.CodePipeline;
+
+  constructor(
+    scope: Construct,
+    id: string,
+    props: ModelDeployCodePipelineStackProps,
+  ) {
+    super(scope, id, props);
+
+    const {
+      projectName,
+      modelPackageGroupName,
+      sagemakerArtifactsBucketName,
+      modelBuildRepositoryName,
+      infraRepo: infraRepoProps,
+      deploymentGroup,
+      modelApprovalTopicName,
+    } = props;
+
+    const sourceBranch = deploymentGroup.sourceBranch;
+    const pipelineName =
+      props.pipelineName || `${projectName}-${sourceBranch}-deploy`;
+    const buildEnv = deploymentGroup.buildEnvironment;
+    const deployEnvironments = deploymentGroup.deployEnvironments || [];
+
+    // add build env to deploy env if not already exists
+    if (!deployEnvironments.find((env) => env.account === buildEnv.account)) {
+      deployEnvironments.unshift(buildEnv);
+    }
+
+    // create SSM param to store the approved model version that will be deployed
+    const approvedModelParam = new ssm.StringParameter(
+      this,
+      'ApprovedModelParam',
+      {
+        parameterName: `/${projectName}/${deploymentGroup.name}/approved-model`,
+        stringValue: 'NONE',
+        description: 'The approved model version to deploy.',
+      },
+    );
+
+    const modelBuildRepo = codecommit.Repository.fromRepositoryName(
+      this,
+      'ModelBuildRepo',
+      modelBuildRepositoryName,
+    );
+
+    const infraRepo = codecommit.Repository.fromRepositoryName(
+      this,
+      'InfraRepo',
+      infraRepoProps.name,
+    );
+
+    const pipeline = new cdk.pipelines.CodePipeline(this, 'Pipeline', {
+      pipelineName,
+      // this pipeline will be updated by project-infra pipeline
+      selfMutation: false,
+      crossAccountKeys: true,
+      artifactBucket: utils.createPipelineArtifactsBucket(this),
+      synth: new cdk.pipelines.CodeBuildStep('Synth', {
+        input: cdk.pipelines.CodePipelineSource.codeCommit(
+          infraRepo,
+          infraRepoProps.branch || DEFAULT_BRANCH,
+          {
+            // trigger the pipeline only when a model is approved
+            // this will be triggered by the lambda function listening on model approval events
+            trigger: cdk.aws_codepipeline_actions.CodeCommitTrigger.NONE,
+          },
+        ),
+        additionalInputs: {
+          modelBuildRepo: cdk.pipelines.CodePipelineSource.codeCommit(
+            modelBuildRepo,
+            sourceBranch,
+            {
+              trigger: cdk.aws_codepipeline_actions.CodeCommitTrigger.NONE,
+            },
+          ),
+        },
+        commands: ['echo hello2', 'npm install', 'npx cdk synth --no-lookups'],
```

Comment:
nit `echo hello2`

Reply:
Good catch. Pushed a new revision removing this.